src/application/common/endpoint.ts

```diff
@@ -24,7 +27,21 @@ export class Endpoint {
     }
 
     protected get host() {
-        return location.host || "127.0.0.1:3000"
+
+
+        if (location.host === "") {
+            this.fs.readFile('port', "utf8", function (err: string, data: any) {
```

Comment:
also you cannot use node APIs in common modules, please review code organization: https://github.com/theia-ide/theia/wiki